File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert num
    st = {
      3 => "Pling",
      5 => "Plang",
      7 => "Plong"
    }.map{ |n, w|
      w if num % n == 0
    }.join

    st.empty? ? num.to_s : st
  end

end

class BookKeeping
  VERSION = 3
end

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
  val factors = mapOf(
    3 to "Pling",
    5 to "Plang",
    7 to "Plong"
  )

  fun convert(number: Int): String {
    val output = factors
      .filterKeys{number % it == 0}
      .values
      .joinToString("")

    return output.takeUnless{it.isBlank()} ?: number.toString()
  }
}
